repos/fdbt-site/tests/pages/api/managePurchaseMethod.test.ts

```diff
@@ -0,0 +1,240 @@
+
+        expect(updateSessionAttributeSpy).toHaveBeenCalledWith(
+            req,
+            GS_PURCHASE_METHOD_ATTRIBUTE,
+            expectedSessionAttributeCall,
+        );
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/managePurchaseMethod',
+        });
+    });
+
+    it('redirects back to /managePurchaseMethod if only one ticketFormats is selected', () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: {
+                ticketFormats: 'Paper Ticket',
+                name: 'a name',
+            },
+        });
+        const expectedSessionAttributeCall: GlobalSettingsAttribute<SalesOfferPackage> = {
+            inputs: {
+                purchaseLocations: [],
+                ticketFormats: ['Paper Ticket'],
+                paymentMethods: [],
+                name: 'a name',
+            },
+            errors: [
+                {
+                    id: 'checkbox-0-on-board',
+                    errorMessage: expect.any(String),
+                },
+                {
+                    id: 'checkbox-0-cash',
+                    errorMessage: expect.any(String),
+                },
+            ],
+        };
+
+        managePurchaseMethod(req, res);
+
+        expect(updateSessionAttributeSpy).toHaveBeenCalledWith(
+            req,
+            GS_PURCHASE_METHOD_ATTRIBUTE,
+            expectedSessionAttributeCall,
+        );
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/managePurchaseMethod',
+        });
+    });
+
+    it('redirects back to /managePurchaseMethod if one selection is missing', () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: {
+                purchaseLocations: ['OnBoard', 'Online Account'],
+                ticketFormats: ['Paper Ticket', 'Debit/Credit card'],
+                name: 'a name',
+            },
+        });
+        const expectedSessionAttributeCall: GlobalSettingsAttribute<SalesOfferPackage> = {
+            inputs: {
+                purchaseLocations: ['OnBoard', 'Online Account'],
+                ticketFormats: ['Paper Ticket', 'Debit/Credit card'],
+                paymentMethods: [],
+                name: 'a name',
+            },
+            errors: [
+                {
+                    id: 'checkbox-0-cash',
+                    errorMessage: expect.any(String),
+                },
+            ],
+        };
+
+        managePurchaseMethod(req, res);
+
+        expect(updateSessionAttributeSpy).toHaveBeenCalledWith(
+            req,
+            GS_PURCHASE_METHOD_ATTRIBUTE,
+            expectedSessionAttributeCall,
+        );
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/managePurchaseMethod',
+        });
+    });
+
+    it('redirects back to /managePurchaseMethod if no name', () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: {
+                name: 'my name',
```

Comment:
Should name be omitted here, and paymentMethods included?

Reply:
Good spot 👍